### ISSUE TITLE
[$35] Error during posting a first comment under a phase by user not in the team

### ISSUE THREAD

**Issue by maxceem** [appirio-tech/connect-app#2772]
### Steps to reproduce the problem
- Using user `pshah_customer` create a new project
- Use user `pshah_manager` to open this new project
- Open **Project Plan** tab, open any phase, click **Discussions** tab and try to post a new comment
- After pressing **Post** comment will be disabled as its being sending but this will never finish.
   There is an error in console `TypeError: Cannot read property 'userId' of undefined`
- If we reload the page, we can see the comment is here. And we can create another comment without any issue

### Screenshot/screencast
https://monosnap.com/file/Xztqqll1vSgLmeePwj4BZ2jHdkEjNM

fyi @RishiRajSahu

**Comment by vikasrohit** [comment]
@maxceem I think we saw similar issue long time back as well where an admin/manager was not able to post when he/she is not part of the team and trying to make first post in that project. It might be related to the user missing in the redux state.
I guess, we  can pick this one in CF, if you think it is error on front end.

**Comment by maxceem** [comment]
@vikasrohit this issue looks like frontend.

I want to note, that when we create the same way the first post on the **Dashboard** page, there is no error, but comment is rendered without user avatar and name. Instead it shows `Connect User`:

![image](https://user-images.githubusercontent.com/146016/51588937-71ca2b00-1f20-11e9-8378-719fda2bd76c.png)

After I reload the page, the same comment is redered properly with a correct name and avatar:

![image](https://user-images.githubusercontent.com/146016/51588981-8e666300-1f20-11e9-8b41-1e1d87412caa.png)

What way do you wanna fix it?
1. Should we make a quick fix and just avoid error for Phase Discussion tab same like it's done on the Dashboard, but showing `Connect User` intead of user real name with no avatar.
2. Or fix it fully in both places:
     - when a user creates a post we have to check if already have information about that user loaded into the Redux store or no (now we load users who are part of the team, or who made some posts)
     - if we don't have user info in the store we have load his info into the store together with making a post, and show post only after both operations are complete: post is created and we loaded info to render user avatar and name properly (as user who is creating a post is currently active logged-in user I think we already have all necessary information about him, and instead of loading his information again, we can copy his information to the place in Redux store where comments can get it)

What do you think?

**Comment by vikasrohit** [comment]
Yes, 2.iii is the way to go. In fact I was under impression that we are already keeping the logged in user's info in the members redux state. So, I guess we just need to copy the member info to the `members` redux state on successful user login and things should work perfectly after that.

**Comment by maxceem** [comment]
To sum up:
- as we make a post from logged-in user, it means that we already have information about the current user
- when we load information about the logged-in user (we do it to show logged user avatar in the top right corner), we also have to copy the data from `loadUser.user` to `members` list so later comment component can take it
   - we should copy required set of fields, not all user data
     
      ![image](https://user-images.githubusercontent.com/146016/51607658-42cbad80-1f50-11e9-90e8-294b3351de52.png)
- this should fix the issue for both places:
   - posts on the **Dashboard** page, where now we see `Connect User` instead of the real name
   - posts inside phases on the **Discussion** tab, where now we have an error when making the first post as shown in the demo video.

**Comment by maxceem** [comment]
This issue is now open for pick up.

**Comment by afrisalyp** [comment]
@maxceem,

Can I work on this ticket?

**Comment by maxceem** [comment]
Sure @afrisalyp

**Comment by afrisalyp** [comment]
Hi, this is ready for review.
Here is the PR https://github.com/appirio-tech/connect-app/pull/2781

The idea is that the members reducer listens "LOAD_USER_SUCCESS" action and copy the user to the members data.
Please review.

Thanks.

**Comment by RishiRajSahu** [comment]
Fixed, working fine in dev.
